### PR TITLE
nixos/httpd: module cleanup

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
First 9 commits are just module cleanup and should be squashed after review. By reading commit by commit it is easy to see I didn't really change anything, just cleanup.
Final commit should remain separate and provides an easy way for users to decide if they want to run `httpd` as a non `root` user by simply adding this to their configuration:
```
systemd.services.httpd = {
  serviceConfig.User = lib.mkForce "wwwrun";
  AmbientCapabilities = "CAP_NET_BIND_SERVICE";
};
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-servers/apache-httpd/default.nix [hunk 1]
```diff
@@ -689,6 +681,12 @@ in
       "access_compat"
     ];
 
+    systemd.tmpfiles.rules =
+      let
+        svc = config.systemd.services.httpd.serviceConfig;
+      in
+        [ "d '${cfg.logDir}' 0700 ${svc.User} ${svc.Group}" ];
```
Comment (flokli) [inline nixos/modules/services/web-servers/apache-httpd/default.nix]: The code previously didn't handle that either, but on a change of `svc.User`/`svc.Group` ownership of created files isn't updated.

We could add a second line with `Z` here to do that, similar to how it's done in the nginx module:
```
"Z '${cfg.logDir}' - ${svc.User} ${svc.Group} - -"
```

**Review by flokli — COMMENTED** [review]
A improvement suggestion, otherwise LGTM.

File: nixos/modules/services/web-servers/apache-httpd/default.nix (hunk 1 above)
Comment (aanderse) [reply nixos/modules/services/web-servers/apache-httpd/default.nix]: I was a bit hesitant to do that... but yeah giving it more thought I believe you are correct. Will update.

Thanks!

**Comment by aanderse** [comment]
@GrahamcOfBorg test limesurvey mediawiki moodle nagios wordpress

**Comment by aanderse** [comment]
Thanks @flokli :tada:

**Pull request merged**